index.ts

```diff
@@ -12,6 +12,7 @@ type UTXO = {
   txid: string;
   vout: number;
   WIF: string;
+  is_taproot: bool;
```

Comment:
btw, what inputs should be _excluded_ from participating in SP?
only `p2pk`, the ones no one uses nowadays anyway?
should we add them explicitly to `UTXO` type?

Reply:
Any UTXO can be used to fund the transaction, but only the following scriptPubKeys are used to do the shared secret derivation:

* P2TR
* P2WPKH
* P2SH-P2WPKH
* P2PKH

I suppose we could require the wallet to pre-filter the `UTXOs` list before passing them to `createTransaction` , or provide a `inputsForSharedSecretDerivation(utxos: UTXO[]): UTXO[]` function which takes in the UTXOs the wallet wants to use to fund the transaction and filters it down to just the UTXOs that will be used for the tweak?

Reply:
Maybe the easiest thing is to add a `bool` to the UTXO type which indicates it's meant to be used for the silent payments tweak, e.g. `is_silent_payment` or something. The wallet then needs to set this bool when doing coin selection because that's when the wallet likely knows the `scriptPubKey` type, either by knowing which descriptor it came from, or by knowing the scriptPubKey directly

Reply:
im thinking about the consumer of a lib. developer should be able to use it without consulting the original spec.
if the param will be just `bool` like 'can_be_used_for_sp', developer will be like, duh, of course i can.
but instead if developer provides proper types of each utxo, the SP lib willthink for him

Reply:
yeah, agree it's better to have the library apply the logic. I think the main question is how we standardize libraries communicating what "type" the UTXO is. not sure if something already exists for this? Ideally we can reuse it

Reply:
this is up to us. it just needs to be present in type definitions, and documentation